aten/src/ATen/TensorIndexing.h

```diff
@@ -247,7 +247,7 @@ static inline Tensor applySelect(
 
     auto size = (*self_sizes)[dim];
     TORCH_CHECK_INDEX(
-        size >= -index && size > index,
+        size > -1 - index && size > index,
```

Comment:
nit: writing this as `-(index + 1)` is clearer in my head but they are equivalent so you can use whichever you prefer.

Reply:
Considering that both of them translate to just `not`, yeah, let me rewrite it: https://godbolt.org/z/3vcnMYf9a

Reply:
Actually, I'm glad I've added a test for `std::numeric_limits<int64_t>::max()`: this optimization screws up symints again, as `-(int64_max+1)`  is `int64_min` again, with the same carry flag problem as before